examples/simple-bot/simple-bot-slack.js

```diff
@@ -14,43 +14,51 @@
  * limitations under the License.
  */
 
-require('dotenv').load();
+require('dotenv').config();
 
-var Botkit = require('botkit');
-var express = require('express');
-var middleware = require('botkit-middleware-watson')({
+const { Botkit } = require('botkit');
+const { MemoryStorage } = require('botbuilder');
+const { SlackAdapter } = require('botbuilder-adapter-slack');
+
+const express = require('express');
+const WatsonMiddleware = require('botkit-middleware-watson').WatsonMiddleware;
+
+const middleware = new WatsonMiddleware({
   iam_apikey: process.env.ASSISTANT_IAM_APIKEY,
-  password: process.env.ASSISTANT_PASSWORD,
   workspace_id: process.env.WORKSPACE_ID,
   url: process.env.ASSISTANT_URL || 'https://gateway.watsonplatform.net/assistant/api',
   version: '2018-07-10'
 });
 
 // Configure your bot.
-var slackController = Botkit.slackbot({ clientSigningSecret: process.env.SLACK_SIGNING_SECRET });
-var slackBot = slackController.spawn({
-  token: process.env.SLACK_TOKEN
+const adapter = new SlackAdapter({
+  clientSigningSecret: process.env.SLACK_CLIENT_SIGNING_SECRET,
+  botToken: process.env.SLACK_TOKEN,
 });
-slackController.hears(['.*'], ['direct_message', 'direct_mention', 'mention'], function(bot, message) {
-  slackController.log('Slack message received');
-  middleware.interpret(bot, message, function() {
-    if (message.watsonError) {
-      console.log(message.watsonError);
-      bot.reply(message, message.watsonError.description || message.watsonError.error);
-    } else if (message.watsonData && 'output' in message.watsonData) {
-      bot.reply(message, message.watsonData.output.text.join('\n'));
-    } else {
-      console.log('Error: received message in unknown format. (Is your connection with Watson Assistant up and running?)');
-      bot.reply(message, 'I\'m sorry, but for technical reasons I can\'t respond to your message');
-    }
-  });
+const controller = new Botkit({
+    adapter,
+    storage: new MemoryStorage(),
+    // ...other options
 });
 
-slackBot.startRTM();
+
+controller.hears(['.*'], ['direct_message', 'direct_mention', 'mention'], async (bot, message) => {
+  console.log('Slack message received');
+  await middleware.interpret(bot, message);
+  if (message.watsonError) {
+    console.log(message.watsonError);
+    await bot.reply(message, message.watsonError.description || message.watsonError.error);
+  } else if (message.watsonData && 'output' in message.watsonData) {
+    await bot.reply(message, message.watsonData.output.text.join('\n'));
```

Comment:
It would be good to demonstrate how to use `output.generic` here.

Anyway, samples aren't part of release, please make a release now and fix documentation later.